include/dlaf/cublas/template_cublas.h

```diff
@@ -0,0 +1,91 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2022, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#ifdef DLAF_WITH_CUDA
+
+#include <cublas_v2.h>
+#include "dlaf/cublas/error.h"
+
+#define DLAF_DECLARE_CUBLAS_OP(Name) \
+  template <typename T>              \
+  struct Name
+
+#define DLAF_DEFINE_CUBLAS_OP(Name, Type, f)                         \
+  template <>                                                        \
+  struct Name<Type> {                                                \
+    template <typename... Args>                                      \
+    static void call(Args&&... args) {                               \
+      DLAF_CUBLAS_CALL(cublas##f##_v2(std::forward<Args>(args)...)); \
+    }                                                                \
+  }
+
+namespace dlaf::cublas {
+
+// Level 1
+DLAF_DECLARE_CUBLAS_OP(Axpy);
+DLAF_DEFINE_CUBLAS_OP(Axpy, float, Saxpy);
+DLAF_DEFINE_CUBLAS_OP(Axpy, double, Daxpy);
+DLAF_DEFINE_CUBLAS_OP(Axpy, std::complex<float>, Caxpy);
+DLAF_DEFINE_CUBLAS_OP(Axpy, std::complex<double>, Zaxpy);
+
+// Level 2
+DLAF_DECLARE_CUBLAS_OP(Gemv);
+DLAF_DEFINE_CUBLAS_OP(Gemv, float, Sgemv);
+DLAF_DEFINE_CUBLAS_OP(Gemv, double, Dgemv);
+DLAF_DEFINE_CUBLAS_OP(Gemv, std::complex<float>, Cgemv);
+DLAF_DEFINE_CUBLAS_OP(Gemv, std::complex<double>, Zgemv);
+
+DLAF_DECLARE_CUBLAS_OP(Trmv);
+DLAF_DEFINE_CUBLAS_OP(Trmv, float, Strmv);
+DLAF_DEFINE_CUBLAS_OP(Trmv, double, Dtrmv);
+DLAF_DEFINE_CUBLAS_OP(Trmv, std::complex<float>, Ctrmv);
+DLAF_DEFINE_CUBLAS_OP(Trmv, std::complex<double>, Ztrmv);
+
+// Level 3
```

Comment:
```suggestion
}

#define DLAF_MAKE_CUBLAS_OP(Name, f) \
DLAF_DECLARE_CUBLAS_OP(Name); \
DLAF_DEFINE_CUBLAS_OP(Name, float, S##f); \
DLAF_DEFINE_CUBLAS_OP(Name, double, D##f); \
DLAF_DEFINE_CUBLAS_OP(Name, std::complex<float>, C##f); \
DLAF_DEFINE_CUBLAS_OP(Name, std::complex<double>, Z##f)

namespace dlaf::cublas {

// Level 1
DLAF_MAKE_CUBLAS_OP(Axpy, axpy);

// Level 2
DLAF_MAKE_CUBLAS_OP(Gemv, gemv);
DLAF_MAKE_CUBLAS_OP(Trmv, trmv);

// Level 3
```

Reply:
I seem to have left out part of my comment, which was that this change would be unrelated and doesn't have to be done here. It just stood out and I wanted to put it on record...

Reply:
Too late... It is done.
BTW: I added a macro for symmetric cublas too. What do you think?